docs/content/en/latest/references/dashboard_component/dashboard_plugins.md

```diff
@@ -64,60 +67,41 @@ plugin is loaded.
-that version of the Dashboard component. Transitively, any dashboards that link with your plugin will be locked to
-that same version of the Dashboard component.
+All GoodData.UI dependencies are injected into the dashboard plugin in their latest version at runtime. So all the new features are available to the dashboard linked to the plugin at the time we roll them out. However, this has also some consequences: in specific cases, it may happen that when we roll out a new dashboard component feature, it can be incompatible with your deployed dashboard plugin even though it won't be a breaking change on API level as the plugins are pretty powerful in the sense of what you can do with them, and we have no control over the deployed plugin code. To mitigate these issues, adhere to the following recommendations.
+
+### Recommendations
+- Use only public dashboard plugins API and GoodData.UI interfaces. Using anything marked as alpha, beta or internal in the plugin code is at your own risk. If you are missing API for your particular use-case, or something you need is not marked as a stable yet, please [get in touch with us](../../../#Home-JoinLearn) and we may consider to update or extend it.
+- Do not manipulate any CSS or DOM structure of the dashboard, rather stick with the [theming](https://www.gooddata.com/docs/cloud/customize-appearance/create-custom-themes/).
+- Create smaller focused plugins, avoid large complicated plugins that are drastically changing the dashboard behavior and look.
+- For more complex changes, consider better suited approaches: [using standalone dashboard component](../../../learn/embed_dashboards/#EmbedDashboards-Dashboardcomponent) or [building your own custom dashboard](../../../learn/embed_dashboards/#EmbedDashboards-Customdashboards).
 
-This means that any changes and improvements that we make to the KPI Dashboards will not be available on the
-dashboards enhanced by the plugin until you upgrade your plugin and build it against the latest version of
-`@gooddata/sdk-ui-dashboard`.
+### Maintenance
 
-When you have more than one plugin linked to the same dashboard, all these plugins are validated with regard to the version of
-the Dashboard component they can be run with. If no version compatible with all the linked plugins on the dashboard is found,
-none of the plugins is displayed on the dashboard, and an error message is logged into the browser console. In this case,
-review the plugins and [adjust their settings](https://github.com/gooddata/gooddata-ui-sdk/blob/master/tools/dashboard-plugin-template/README.template.md#how-can-i-setup-compatibility-of-the-plugin) so that they can run on the same version of the Dashboard component.
+As the dashboard plugins are mostly integrated with our SaaS platform (GoodData Platform or GoodData Cloud) that is ever evolving, sometimes there is a need to introduce breaking-changes. 
```

Comment:
```suggestion
Since dashboard plugins are primarily integrated with our ever-evolving SaaS platform (GoodData Platform or GoodData Cloud), there are occasions when introducing breaking changes is necessary.
```